src/app/page.tsx

```diff
@@ -64,73 +55,27 @@ export default function Home() {
 }
 
 const SonarEntityPanel = () => {
-  const client = useSonarClient();
-  const { authenticated, ready } = useSonarAuth();
-  const { address, isConnected } = useAccount();
-
-  const [entity, setEntity] = useState<ResourceState<EntityDetails>>({
-    hasFetched: false,
-    loading: false,
-    value: undefined,
+  const { state, loading, entity, error } = useSonarWagmi({
+    saleUUID: sonarConfig.saleUUID,
   });
 
-  const readEntity = useCallback(async () => {
-    if (!ready || !authenticated || !address || !isConnected) {
-      return;
-    }
-
-    try {
-      const entity = await client.readEntity({
-        saleUUID: sonarConfig.saleUUID,
-        walletAddress: address,
-      });
-      setEntity({
-        hasFetched: true,
-        loading: false,
-        value: entity.Entity,
-      });
-    } catch (error) {
-      setEntity({
-        hasFetched: true,
-        loading: false,
-        error: error instanceof Error ? error : new Error(String(error)),
-      });
-    }
-  }, [ready, authenticated, client, address, isConnected]);
-
-  // Read entity on mount
-  useEffect(() => {
-    readEntity();
-  }, [readEntity]);
-
-  // Reset entity state if a user is not authenticated or doesn't have a connected wallet
-  useEffect(() => {
-    if (ready && (!authenticated || !address || !isConnected)) {
-      setEntity({
-        hasFetched: false,
-        loading: false,
-        value: undefined,
-        error: undefined,
-      });
-    }
-  }, [ready, authenticated, address, isConnected]);
-
-  // Coordinate the state of both the wallet and Sonar account being connected.
-  const fullyConnected = ready && authenticated && isConnected;
-  const hasTriedLoading =
-    !entity.loading &&
-    (entity.value !== undefined || entity.error !== undefined);
+  if (state === ConnectionState.Unknown) {
+    return <p>Unknown state</p>;
+  }
 
-  if (!fullyConnected) {
+  if (
+    state === ConnectionState.WalletDisconnected ||
+    state === ConnectionState.SonarDisconnected
+  ) {
     return <p>Connect your wallet and Sonar account to continue</p>;
   }
 
-  if (!ready || entity.loading || !hasTriedLoading) {
+  if (loading) {
     return <p>Loading...</p>;
   }
 
-  if (entity.error) {
-    if (entity.error instanceof APIError && entity.error.status === 404) {
+  if (error) {
+    if (error instanceof APIError && error.status === 404) {
```

Comment:
What about handling this case in the hook, and then the caller can check for `entity.hasFetched && !entity.value`